manifests/init.pp

```diff
@@ -123,50 +123,50 @@
-  Optional[Variant[Boolean,Enum['yes', 'opportunistic', 'no']]] $dnsovertls,
-  Optional[Variant[Boolean,Enum['no-negative']]]         $cache,
-  Optional[Variant[Boolean,Enum['udp','tcp']]]           $dns_stub_listener,
-  Boolean                                                $use_stub_resolver,
-  Boolean                                                $manage_networkd,
-  Enum['stopped','running']                              $networkd_ensure,
-  Boolean                                                $manage_timesyncd,
-  Enum['stopped','running']                              $timesyncd_ensure,
-  Optional[Variant[Array,String]]                        $ntp_server,
-  Optional[Variant[Array,String]]                        $fallback_ntp_server,
-  Boolean                                                $manage_accounting,
-  Hash[String,String]                                    $accounting,
-  Boolean                                                $purge_dropin_dirs,
-  Boolean                                                $manage_journald,
-  Systemd::JournaldSettings                              $journald_settings,
-  Boolean                                                $manage_udevd,
-  Optional[Variant[Integer,String]]                      $udev_log,
-  Optional[Integer]                                      $udev_children_max,
-  Optional[Integer]                                      $udev_exec_delay,
-  Optional[Integer]                                      $udev_event_timeout,
-  Optional[Enum['early', 'late', 'never']]               $udev_resolve_names,
-  Optional[Variant[Integer,String]]                      $udev_timeout_signal,
-  Boolean                                                $manage_logind,
-  Systemd::LogindSettings                                $logind_settings,
-  Boolean                                                $manage_all_network_files,
-  Stdlib::Absolutepath                                   $network_path,
-  Hash                                                   $loginctl_users = {},
-  Hash                                                   $dropin_files = {},
-  Hash                                                   $udev_rules = {},
+  Hash[String,Hash[String, Any]]                      $service_limits = {},
+  Boolean                                             $manage_resolved = false,
+  Enum['stopped','running']                           $resolved_ensure = 'running',
+  Optional[Variant[Array[String],String]]             $dns = undef,
+  Optional[Variant[Array[String],String]]             $fallback_dns = undef,
+  Optional[Variant[Array[String],String]]             $domains = undef,
+  Optional[Variant[Boolean,Enum['resolve']]]          $llmnr = undef,
+  Optional[Variant[Boolean,Enum['resolve']]]          $multicast_dns = undef,
+  Optional[Variant[Boolean,Enum['allow-downgrade']]]  $dnssec = undef,
+  Variant[Boolean,Enum['yes', 'opportunistic', 'no']] $dnsovertls = false,
```

Comment:
I changed the datatype of $dnsovertls and $cache. It was Optional but the common.yaml set both to false.